### ISSUE TITLE
Null check operator used on a null value

### ISSUE THREAD

**Issue by alexaung** [jb3rndt/PersistentBottomNavBarV2#39]
Hi 

I got this error in firebase crashlytics report. But there is no issue on device.

> Fatal Exception: FlutterError
> Null check operator used on a null value. Error thrown Instance of 'ErrorDescription'.
> 
> persistent-tab-view.widget.dart - Line 744
> _PersistentTabViewState.popAllScreens + 744

Device
Model:iPhone 6
Orientation: Portrait
RAM free: 129.84 MB
Disk free: 30.77 GB

Operating system
Version:12.5.5
Orientation: Portrait
Jailbroken:No

Crash
Date:14 Jun 2022, 21:09:04
App version:2.0.1 (28)

**Comment by jb3rndt** [comment]
Yes it uses a the `!` nullcheck operator. Can you send me your `routeAndNavigatorSettings` of every `PersistentNavBarItem` or the `PersistentTabView.custom` if you are using a custom NavBar?

**Comment by alexaung** [comment]
Here is my root_sreen.dart
```
import 'dart:io';
import 'package:flutter/material.dart';
import 'package:flutter_bloc/flutter_bloc.dart';
import 'package:new_version/new_version.dart';
import 'package:persistent_bottom_nav_bar_v2/persistent-tab-view.dart';
import 'package:thitsarparami/blocs/bloc.dart';
import 'package:thitsarparami/connectivity_wrapper/connectivity_wrapper.dart';
import 'package:thitsarparami/helper/constants.dart';
import 'package:thitsarparami/helper/push_notifications_manager.dart';
import 'package:thitsarparami/settings/preferences.dart';
import 'package:thitsarparami/ui/album/album_screen.dart';
import 'package:thitsarparami/ui/audio_service/now_playing_screen.dart';
import 'package:thitsarparami/ui/chanting/chanting_catalog_screen.dart';
import 'package:thitsarparami/ui/chanting/chanting_detial_screen.dart';
import 'package:thitsarparami/ui/chanting/chanting_screen.dart';
import 'package:thitsarparami/ui/home/home_screen.dart';
import 'package:thitsarparami/ui/library/library_screen.dart';
import 'package:thitsarparami/ui/monk/monk_screen.dart';
import 'package:thitsarparami/ui/more/more_screen.dart';
import 'package:thitsarparami/ui/search/search_screen.dart';
import 'package:thitsarparami/ui/setting/setting_screen.dart';
import 'package:thitsarparami/ui/song/song_screen.dart';

import '../../helper/enum.dart';
import '../../widgets/custom_button.dart';

BuildContext? testContext;

class RootScreen extends StatefulWidget {
  static const routeName = '/';
  final BuildContext menuScreenContext;
  const RootScreen({Key? key, required this.menuScreenContext})
      : super(key: key);

  @override
  RootScreenState createState() => RootScreenState();
}

class RootScreenState extends State<RootScreen> {
  PersistentTabController? _controller;
  bool _hideNavBar = false;
  @override
  void initState() {
    super.initState();

    PushNotificationsManager().init();
    _controller = PersistentTabController(initialIndex: 0);
    _hideNavBar = false;
    _loadInitialData();
  }

  _loadInitialData() async {
    if (await ConnectivityWrapper.instance.isConnected) {
      _loadAPKUrl();
      _versionCheck();
      _loadTheme();
    }
  }

  _loadAPKUrl() async {
    List<String> codes = [
      systemDataCodeToString(SystemDataCode.apk),
    ];
    BlocProvider.of<SystemDataBloc>(context).add(GetApkEvent(codes));
  }

  _loadTheme() async {
    BlocProvider.of<ThemeBloc>(context)
        .add(ThemeEvent(appTheme: Preferences.getTheme()));
  }

  _versionCheck() {
    // Instantiate NewVersion manager object (Using GCP Console app as example)
    final newVersion = NewVersion(
      iOSId: 'org.thitsarparami.app',
      androidId: 'org.thitsarparami.app',
    );

    // You can let the plugin handle fetching the status and showing a dialog,
    // or you can fetch the status and display your own dialog, or no dialog.
    const simpleBehavior = false;

    if (simpleBehavior) {
      basicStatusCheck(newVersion);
    } else {
      advancedStatusCheck(newVersion);
    }
  }

  basicStatusCheck(NewVersion newVersion) {
    newVersion.showAlertIfNecessary(context: context);
  }

  // Future<void> openWithStore() async {
  //   try {
  //     StoreLauncher.openWithStore("org.thitsarparami.app").catchError((e) {
  //       if (kDebugMode) {
  //         print('ERROR> $e');
  //       }
  //     });
  //   } on Exception catch (e) {
  //     if (kDebugMode) {
  //       print('$e');
  //     }
  //   }
  // }

  advancedStatusCheck(NewVersion newVersion) async {
    final status = await newVersion.getVersionStatus();
    if (status != null && status.canUpdate) {
      debugPrint(status.releaseNotes);
      debugPrint(status.appStoreLink);
      debugPrint(status.localVersion);
      debugPrint(status.storeVersion);
      debugPrint(status.canUpdate.toString());
      // newVersion.showUpdateDialog(
      //   context: context,
      //   versionStatus: status,
      //   dialogTitle: 'Update Available',
      //   dialogText: 'You can now update this app from ${status.localVersion} to ${status.storeVersion}',
      // );

      showDialog(
        context: context,
        builder: (context) => BlocBuilder<SystemDataBloc, SystemDataState>(
          builder: (context, state) {
            if (state is ApkDataLoaded) {
              return AlertDialog(
                title: const Text("Update Available"),
                content: Text(
                    'You can now update this app from ${status.localVersion} to ${status.storeVersion}'),
                actions: [
                  Platform.isIOS
                      ? Center(
                          child: Column(
                            children: [
                              CustomButton(
                                onPressed: () {
                                  launchURL(status.appStoreLink);
                                },
                                child: Row(
                                  mainAxisAlignment: MainAxisAlignment.center,
                                  crossAxisAlignment: CrossAxisAlignment.center,
                                  children: const [
                                    Icon(Icons.apple),
                                    Padding(
                                      padding: EdgeInsets.all(8.0),
                                      child: Text("Playstore"),
                                    ),
                                  ],
                                ),
                              ),
                            ],
                          ),
                        )
                      : Center(
                          child: Column(
                            children: [
                              CustomButton(
                                onPressed: () {
                                  launchURL(status.appStoreLink);
                                },
                                child: Row(
                                  mainAxisAlignment: MainAxisAlignment.center,
                                  crossAxisAlignment: CrossAxisAlignment.center,
                                  children: const [
                                    Icon(Icons.android),
                                    Padding(
                                      padding: EdgeInsets.all(8.0),
                                      child: Text("Playstore"),
                                    ),
                                  ],
                                ),
                              ),
                              const SizedBox(
                                height: 20,
                              ),
                              CustomButton(
                                onPressed: () {
                                  launchURL(state.systemData.apk!);
                                },
                                child: Row(
                                  mainAxisAlignment: MainAxisAlignment.center,
                                  crossAxisAlignment: CrossAxisAlignment.center,
                                  children: const [
                                    Icon(Icons.cloud_download),
                                    Padding(
                                      padding: EdgeInsets.all(8.0),
                                      child: Text("Download"),
                                    ),
                                  ],
                                ),
                              ),
                              const SizedBox(
                                height: 20,
                              ),
                            ],
                          ),
                        )
                ],
              );
            }
            return Container();
          },
        ),
      );
    }
  }

  List<Widget> _buildScreens() {
    return [
      HomeScreen(
        menuScreenContext: widget.menuScreenContext,
        hideStatus: _hideNavBar,
        onScreenHideButtonPressed: () {
          setState(() {
            _hideNavBar = !_hideNavBar;
          });
        },
      ),
      SearchScreen(
        menuScreenContext: widget.menuScreenContext,
        hideStatus: _hideNavBar,
        tabController: _controller,
        onScreenHideButtonPressed: () {
          setState(() {
            _hideNavBar = !_hideNavBar;
          });
        },
      ),
      LibraryScreen(
        menuScreenContext: widget.menuScreenContext,
        hideStatus: _hideNavBar,
        tabController: _controller,
        onScreenHideButtonPressed: () {
          setState(() {
            _hideNavBar = !_hideNavBar;
          });
        },
      ),
      NowPlayingScreen(
        menuScreenContext: widget.menuScreenContext,
        hideStatus: _hideNavBar,
        onScreenHideButtonPressed: () {
          setState(() {
            _hideNavBar = !_hideNavBar;
          });
        },
        hideCloseButton: true,
      ),
      MoreScreen(
        menuScreenContext: widget.menuScreenContext,
        hideStatus: _hideNavBar,
        tabController: _controller,
        onScreenHideButtonPressed: () {
          setState(() {
            _hideNavBar = !_hideNavBar;
          });
        },
      ),
    ];
  }

  List<PersistentBottomNavBarItem> _navBarsItems() {
    return [
      PersistentBottomNavBarItem(
        icon: const Icon(Icons.home_outlined),
        title: "Home",
        activeColorPrimary:
            Theme.of(context).bottomNavigationBarTheme.selectedItemColor!,
        inactiveColorPrimary:
            Theme.of(context).bottomNavigationBarTheme.unselectedItemColor,
        inactiveColorSecondary: Theme.of(context).primaryColorLight,
        routeAndNavigatorSettings: RouteAndNavigatorSettings(
          initialRoute: '/',
          routes: {
            '/monk': (context) => const MonkScreen(),
            '/album': (context) => const AlbumScreen(),
            '/song': (context) => const SongScreen(),
            '/chanting_catalog': (context) => const ChantingCatalogScreen(),
            '/chanting': (context) => const ChantingScreen(),
            '/chanting_detail': (context) => const ChantingDetailScreen(),
          },
        ),
      ),
      PersistentBottomNavBarItem(
        icon: const Icon(Icons.search),
        title: "Search",
        activeColorPrimary:
            Theme.of(context).bottomNavigationBarTheme.selectedItemColor!,
        inactiveColorPrimary:
            Theme.of(context).bottomNavigationBarTheme.unselectedItemColor,
        inactiveColorSecondary: Theme.of(context).primaryColorLight,
        // routeAndNavigatorSettings: RouteAndNavigatorSettings(
        //   initialRoute: '/',
        //   routes: {
        //     '/monk': (context) =>  const MonkScreen(),
        //     '/album': (context) => const AlbumScreen(),
        //     '/song': (context) => const SongScreen(),
        //     '/chanting_catalog': (context) => const ChantingCatalogScreen(),
        //     '/chanting': (context) => const ChantingScreen(),
        //     '/chanting_detail': (context) => const ChantingDetailScreen(),
        //   },
        // ),
      ),
      PersistentBottomNavBarItem(
        icon: const Icon(Icons.create_new_folder_outlined),
        title: ("Library"),
        activeColorPrimary:
            Theme.of(context).bottomNavigationBarTheme.selectedItemColor!,
        inactiveColorPrimary:
            Theme.of(context).bottomNavigationBarTheme.unselectedItemColor,
        inactiveColorSecondary: Theme.of(context).primaryColorLight,
        // routeAndNavigatorSettings: RouteAndNavigatorSettings(
        //   initialRoute: '/',
        //   routes: {
        //     '/monk': (context) => const MonkScreen(),
        //     '/album': (context) => const AlbumScreen(),
        //   },
        // ),
      ),
      PersistentBottomNavBarItem(
        icon: const Icon(Icons.play_circle_outlined),
        title: ("Play"),
        activeColorPrimary:
            Theme.of(context).bottomNavigationBarTheme.selectedItemColor!,
        inactiveColorPrimary:
            Theme.of(context).bottomNavigationBarTheme.unselectedItemColor,
        inactiveColorSecondary: Theme.of(context).primaryColorLight,
        // routeAndNavigatorSettings: RouteAndNavigatorSettings(
        //   initialRoute: '/',
        //   routes: {
        //     '/first': (context) => MainScreen2(),
        //     '/second': (context) => MainScreen3(),
        //   },
        // ),
        // onPressed: (context) {
        //   pushDynamicScreen(context,
        //       screen: SampleModalScreen(), withNavBar: true);
        // },
      ),
      PersistentBottomNavBarItem(
        icon: const Icon(Icons.more_horiz_outlined),
        title: ("More"),
        activeColorPrimary:
            Theme.of(context).bottomNavigationBarTheme.selectedItemColor!,
        inactiveColorPrimary:
            Theme.of(context).bottomNavigationBarTheme.unselectedItemColor,
        inactiveColorSecondary: Theme.of(context).primaryColorLight,
        routeAndNavigatorSettings: RouteAndNavigatorSettings(
          initialRoute: '/',
          routes: {
            '/settings': (context) => const SettingScreen(),
          },
        ),
      ),
    ];
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.transparent,
      body: PersistentTabView(
        context,
        controller: _controller,
        screens: _buildScreens(),
        items: _navBarsItems(),
        confineInSafeArea: true,
        backgroundColor: Theme.of(context).backgroundColor,
        handleAndroidBackButtonPress: true,
        resizeToAvoidBottomInset: true,
        stateManagement: true,
        navBarHeight: MediaQuery.of(context).viewInsets.bottom > 0
            ? 0.0
            : kBottomNavigationBarHeight,
        hideNavigationBarWhenKeyboardShows: true,
        margin: const EdgeInsets.all(0.0),
        popActionScreens: PopActionScreensType.all,
        bottomScreenMargin: 0.0,
        selectedTabScreenContext: (context) {
          testContext = context;
        },
        hideNavigationBar: _hideNavBar,
        decoration: const NavBarDecoration(
          border: Border(
              top: BorderSide(
                  color: Colors.grey, width: 0.2, style: BorderStyle.solid)),
        ),
        popAllScreensOnTapOfSelectedTab: true,
        itemAnimationProperties: const ItemAnimationProperties(
          duration: Duration(milliseconds: 400),
          curve: Curves.ease,
        ),
        screenTransitionAnimation: const ScreenTransitionAnimation(
          animateTabTransition: true,
          curve: Curves.easeInOut,
          duration: Duration(milliseconds: 500),
        ),
        navBarStyle:
            NavBarStyle.style1, // Choose the nav bar style with this property
      ),
    );
  }
}

```

**Comment by jb3rndt** [comment]
Hm I cant find the reason why this happens, does it happen very often?

**Comment by alexaung** [comment]
I am calling pdf and youtube screen without NavBar. May It be because of that?

**Comment by jb3rndt** [comment]
I cannot reproduce the issue still, sorry... You said that it happens in crashlytics, so does it also happen to you in debug mode or while developing? I think it would be strange if it only happens in release mode and is observed by crashlytics.

**Comment by alexaung** [comment]
I can't reproduce in debug as well. 
![persisten-tab-view widget dart](https://user-images.githubusercontent.com/1129760/181524231-4e3febc0-4373-4fe9-ab6e-9ff5e8a85581.png)

**Comment by jb3rndt** [comment]
I actually noticed that issue as well in crashlytics lately. It was only on iOS and only while the app was in background. So I just added a check that prevents the screens from popping when the context is null. Its available in version 4.2.8.
